src/services/consent/utils.js

```diff
@@ -0,0 +1,321 @@
+									],
+								},
+								{
+									$and: [
+										{ $gt: ['$birthday', firstLevel] },
+										{ $eq: ['$consent.parentConsents.privacyConsent', true] },
+										{ $eq: ['$consent.parentConsents.termsOfUseConsent', true] },
+									],
+								},
+
+							],
+						},
+						then: 'ok',
+
+					},
+					{
+						case: {
+							$and: [
+								{ $gt: ['$birthday', secondLevel] },
+								{ $lte: ['$birthday', firstLevel] },
+								{ $eq: ['$consent.parentConsents.privacyConsent', true] },
+								{ $eq: ['$consent.parentConsents.termsOfUseConsent', true] },
+							],
+						},
+						then: 'parentsAgreed',
+					},
+				],
+				default: 'missing',
+			},
+		};
+	}
+
+	const aggregation = [];
+
+	if (match) {
+		aggregation.push({
+			$match: match,
+		});
+	}
+
+	if (select) {
+		aggregation.push({
+			$project: project,
+		});
+	}
+
+	if (consentStatus) {
+		let list = [];
+		if (Array.isArray(consentStatus)) {
+			list = consentStatus;
+		} else if (Array.isArray(consentStatus.$in)) {
+			list = consentStatus.$in;
+		} else if (typeof consentStatus === 'string') {
+			list = [consentStatus];
+		}
+
+		aggregation.push({
+			$match: {
+				consentStatus: {
+					$in: list,
+				},
+			},
+		});
+		/* aggregation.push({
+			match: {
+				$in: ['$consentStatus', list],
+			},
+		}); */
+	}
+
+	if (sort) {
+		let mSort = sort;
+
+		if (Array.isArray(sort) && sort.includes('consentStatus')) {
+			const i = sort.indexOf('consentStatus');
+			mSort = sort.splice(i, 1, 'consentSortParam');
+			stageAddConsentSortParam(aggregation);
+		} else if (typeof sort === 'object' && ({}).hasOwnProperty.call(sort, 'consentStatus')) {
+			sort.consentSortParam = sort.consentStatus;
+			delete sort.consentStatus;
+			stageAddConsentSortParam(aggregation);
+		}
+		aggregation.push({
+			$sort: mSort,
+		});
+	}
+
+	aggregation.push({
+		$facet: {
+			data: [
+				{
+					$skip: skip,
+				},
+				{
+					$limit: limit,
+				},
+			],
+			total: [{
+				$count: 'count',
+			}],
+		},
+	});
+
+	aggregation.push({
+		$project: {
+			total: {
+				$reduce: {
+					input: '$total',
+					initialValue: 0,
+					in: { $add: ['$$value', '$$this.count'] },
+				},
+			},
+			data: true,
+		},
+	});
+
+	aggregation.push({
+		$addFields: {
+			limit,
+			skip,
+		},
+	});
+
+
+	return aggregation;
+};
+
+
+const createConsentFilterQuery = (...status) => {
+	const currentDate = new Date();
+	const secondConsentSwitchDate = new Date();
+	secondConsentSwitchDate.setFullYear(currentDate.getFullYear() - 16); // TODO: get age from default.conf
+	const firstConsentSwitchDate = new Date();
+	firstConsentSwitchDate.setFullYear(currentDate.getFullYear() - 14);
+
+	const requiredConsents = {
+		privacyConsent: true,
+		termsOfUseConsent: true,
+	};
+
+	const missingConsents = {
+		$or: [
+			{ privacyConsent: false },
+			{ termsOfUseConsent: false },
+		],
+	};
+
+	const orConditions = status.reduce((query, status) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected to return a value in arrow function.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5276619)